### ISSUE TITLE
Incorrect RCS1151 Remove redundant cast

### ISSUE THREAD

**Issue by mikernet** [JosefPihrt/Roslynator#611]
**Product and Version Used**: 
Roslynator 2019 v2.2.0

**Steps to Reproduce**:
```csharp
object value = GetItems();
bool hasAnyItems = ((IEnumerable)value).Cast<object>().Any(); // triggers RCS1151
```
**Actual Behavior**:
Cast to `IEnumerable` triggers RCS1151

**Expected Behavior**:
Removing the cast is not valid so RCS1151 should not be triggered.
